apps/dashboard/lib/auth/workos.ts

```diff
@@ -84,46 +92,83 @@ export class WorkOSAuthProvider extends BaseAuthProvider {
     }
   }
 
-  async refreshSession(sessionToken: string | null): Promise<SessionRefreshResult> {
-    if (!sessionToken) {
-      throw new Error("No session token provided");
+  async refreshAccessToken(currentRefreshToken: string | null): Promise<SessionRefreshResult> {
+    if (!currentRefreshToken) {
+      throw new Error("No refresh token provided");
     }
 
-    try {
-      const session = this.provider.userManagement.loadSealedSession({
-        sessionData: sessionToken,
-        cookiePassword: this.cookiePassword,
-      });
-
-      const refreshResult = await session.refresh({
-        cookiePassword: this.cookiePassword,
-      });
+    const expiresAt = new Date();
+    expiresAt.setDate(expiresAt.getDate() + 7);
 
-      if (refreshResult.authenticated && refreshResult.session) {
-        // Set expiration to 7 days from now
-        const expiresAt = new Date();
-        expiresAt.setDate(expiresAt.getDate() + 7);
-
-        return {
-          newToken: refreshResult.sealedSession!,
-          expiresAt,
+    try {
+      const { sealedSession, refreshToken, accessToken, user, organizationId } =
+        await this.provider.userManagement.authenticateWithRefreshToken({
+          clientId: this.clientId,
+          refreshToken: currentRefreshToken,
           session: {
-            userId: refreshResult.session.user.id,
-            orgId: refreshResult.session.organizationId ?? null,
+            sealSession: true,
+            cookiePassword: this.cookiePassword,
           },
-        };
-      }
+        });
 
-      throw new Error("reason" in refreshResult ? refreshResult.reason : "Session refresh failed");
+      return {
+        expiresAt,
+        sessionToken: sealedSession!,
+        accessToken,
+        refreshToken,
+        session: {
+          userId: user.id,
+          orgId: organizationId ?? null,
+          role: null, // not returned in authenticateWithRefreshToken
+        },
+      };
     } catch (error) {
-      console.error("Session refresh error:", {
-        error: error instanceof Error ? error.message : "Unknown error",
-        token: sessionToken ? `${sessionToken.substring(0, 10)}...` : "no token",
-      });
+      console.error("Access token failed to refresh: ", error);
       throw error;
     }
   }
 
+  //   if (!sessionToken) {
```

Comment:
Let's yeet if we don't need it any more... will leave to your judgement

Reply:
somehow missed this